apps/api/v2/src/ee/bookings/controllers/bookings.controller.ts

```diff
@@ -30,7 +42,51 @@ import { ApiResponse } from "@calcom/platform-types";
 export class BookingsController {
   private readonly logger = new Logger("ee bookings controller");
 
-  constructor(private readonly oAuthFlowService: OAuthFlowService) {}
+  constructor(
+    private readonly oAuthFlowService: OAuthFlowService,
+    private readonly prismaReadService: PrismaReadService
+  ) {}
+
+  // note(Rajiv): currently this endpoint is atoms only
+  @Get("/")
+  @UseGuards(AccessTokenGuard)
+  async getBookings(
+    @GetUser() user: User,
+    @Query() queryParams: GetBookingsInput
+  ): Promise<ApiResponse<unknown>> {
+    const { filters, cursor, limit } = queryParams;
+    const bookings = await getAllUserBookings({
+      bookingListingByStatus: filters.status,
+      skip: cursor ?? 0,
+      take: limit ?? 10,
+      filters: filters,
+      ctx: {
+        user: { email: user.email, id: user.id },
+        prisma: this.prismaReadService.prisma as unknown as PrismaClient,
+      },
+    });
+
+    return {
+      status: SUCCESS_STATUS,
+      data: bookings,
+    };
+  }
+
+  // note(Rajiv): currently this endpoint is atoms only
+  @Get("/:bookingUid")
+  @UseGuards(AccessTokenGuard)
```

Comment:
I think this endpoint should not be behind access token guard, for example on app.cal.com, you can access booking success page of any booking without being auth.

for example open this link in an incognito tab: https://app.cal.com/booking/2JeoZEPp6GB7uCo1xTLbLE?allRemainingBookings=false&email=morgan%40cal.com

Reply:
Right, makes sense. After a booking has been made there shouldn't be a need to use access token to fetch that booking details, will fix this.